### PR TITLE
feature(coq): coq_version variable

### PR DESCRIPTION
**Pull request by rgrinberg**
- [x] `coq_version` 
- [ ] `coq_config` macro
- [ ] documentation
- [ ] CHANGES
- [ ] Tests

### CODE REVIEW

File: src/dune_lang/pform.ml [hunk 1]
```diff
@@ -46,6 +46,7 @@ module Var = struct
     | Corrected_suffix
     | Inline_tests
     | Toolchain
+    | Coq_version
```
Comment (ejgallego) [inline src/dune_lang/pform.ml:49]: I guess we could add a parameter to this to have a `%{coq:var}` form, WDYT?

File: src/dune_rules/coq_config.ml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+open Import
+open Memo.O
+
+let impl bin =
+  let* _ = Build_system.build_file bin in
+  let+ line =
+    Memo.of_non_reproducible_fiber
```
Comment (ejgallego) [inline src/dune_rules/coq_config.ml:7]: Does this mean that even within a single run we will re-exec?

All Coq config is deterministic, just set a Coq configure time.

Comment (ejgallego) [reply src/dune_rules/coq_config.ml:7]: And guarantee to change only if the hash changes.

**Comment by Alizter** [comment]
FTR I am still working on the coq-config thing, but I will be unable to work on it much this week.

**Pull request closed without merging**